### PR TITLE
terraform: Fix all `tfsec` and `ansible-lint` warnings within `examples/aws/terraform`

### PR DESCRIPTION
**Pull request by reedloden**
There were numerous `tfsec` and `ansible-lint` warnings within the `examples/aws/terraform` directory. Address all issues, adding ignores where needed to bring warning count down to zero.

### CODE REVIEW

**Review by wadells — APPROVED** [review]
LGTM.

Context-wise: I'm not sure that this is maintained.  I think it is maybe replaced by https://github.com/gravitational/teleport-cluster-terraform.  @webvictim or @hugoShaka  would know better.

**Comment by hugoShaka** [comment]
> Context-wise: I'm not sure that this is maintained. I think it is maybe replaced by https://github.com/gravitational/teleport-cluster-terraform. @webvictim or @hugoShaka would know better.

image


Joke aside, I was not aware we had a terraform module in a standalone repo. I always worked with `examples/aws/terraform` as this is the one [linked to in the docs](https://goteleport.com/docs/deploy-a-cluster/deployments/aws-terraform/).

I will see with @r0mant and @webvictim what's the current state and where do we want to go because the code seems duplicated and this is confusing.

File: examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml
Comment (hugoShaka) [inline examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml:28]: I think this is broken as `src` and `dest` should be children of `copy`.

File: examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml
Comment (hugoShaka) [inline examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml:30]: idem

File: examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml
Comment (webvictim) [inline examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml:26]: Indentation is funky here. We should either outdent the `copy` or indent the `src` and `dest` for clarity IMO.

File: examples/aws/terraform/starter-cluster/cluster_sg.tf
Comment (webvictim) [inline examples/aws/terraform/starter-cluster/cluster_sg.tf:65]: I don't quite understand why some of these changes in the diff are removing extra tabs while some others are adding them in. Automatic rather than deliberate?

Comment (reedloden) [reply examples/aws/terraform/starter-cluster/cluster_sg.tf:65]: Yeah, that's from `terraform fmt`. terraform has its own preferred style that it likes to enforce (and the linting jobs enforce it as well).

**Comment by hugoShaka** [comment]
[The related issue is here](https://github.com/gravitational/teleport/issues/16553), it's waiting for Product team approval and likely won't be tackled until late Q4. In the meantime we'll have to watch terraform PRs and make sure both repos stay aligned as much as possible.

File: examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml
Comment (reedloden) [reply examples/aws/terraform/ha-autoscale-cluster/ansible/upgrade.yaml:26]: I did the latter. `ansible-lint` still passes.

**Review by hugoShaka — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
